engine/src/test/scala/cromwell/services/ServiceRegistryActorSpec.scala

```diff
@@ -0,0 +1,46 @@
+package cromwell.services
+
+import akka.pattern.ask
+import com.typesafe.config.ConfigFactory
+import cromwell.engine.WorkflowSourceFiles
+import cromwell.services.KeyValueServiceActor._
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.{FlatSpec, Matchers}
+
+class ServiceRegistryActorSpec extends FlatSpec with Matchers with ScalaFutures with CromwellServicesTest {
+
+  val cromwellConfig = ConfigFactory.parseString(
+    s"""services: {
+        |  KeyValue: {
+        |    class: "cromwell.services.KeyValueServiceActor"
+        |  }
+        |}
+     """.stripMargin
+  )
+
+  val sources = WorkflowSourceFiles(
+    wdlSource="task a {command{}} workflow w {call a}",
+    inputsJson="{}",
+    workflowOptionsJson="{}"
+  )
+
+  it should "insert a key/value" in {
+    val serviceRegistryActor = makeServiceRegistryActor(dataAccess, cromwellConfig)
+    val descriptor = makeWorkflowDescriptor(sources)
+    val callKey = getBackendJobDescriptorKey(descriptor, "a")
+    val kvPut = KvPut(KvPair(ScopedKey(callKey, "k"), Option("v")))
+    val kvGet = KvGet(ScopedKey(callKey, "k"))
+
+    val future = for {
+      _ <- dataAccess.createWorkflow(descriptor, Nil, descriptor.namespace.workflow.calls, localBackend)
+      r2 <- ask(serviceRegistryActor, kvGet).mapTo[KvKeyLookupFailed]
+      _ = r2.action shouldEqual kvGet
+      r0 <- ask(serviceRegistryActor, kvPut).mapTo[KvResponse]
+      _ = r0 shouldEqual KvPutSuccess(kvPut)
+      r1 <- ask(serviceRegistryActor, kvGet).mapTo[KvResponse]
+      _ = r1 shouldEqual KvPair(ScopedKey(callKey, "k"), Option("v"))
```

Comment:
Not a huge deal.

The `r2`, `r0`, `r1` names aren't descriptive. But I'm more curious (and others may in the [future](http://stackoverflow.com/questions/876089/who-wrote-this-programing-saying-always-code-as-if-the-guy-who-ends-up-maintai))… why the 2,0,1 order?